### ISSUE TITLE
Toggling react.enable flag to false breaks application

### ISSUE THREAD

**Issue by tepi** [vaadin/flow#19951]
### Description of the bug

Just did a quick test with simple Flow+SpringBoot application downloaded from start.vaadin.com.

First I ran the application as-is, it runs and confirmed react-router is used.

Then I set `vaadin.react.enable=false` in `application.properties` and restarted the application. This leads into exception in vite:

```
ERROR(TypeScript)  Cannot find module 'react-router-dom' or its corresponding type declarations.
FILE  /Users/tepi/Downloads/skeleton-starter-flow-spring-24 2/src/main/frontend/generated/vaadin.ts:42:24

   40 | applyTheme(document);
   41 |
 > 42 | import { Outlet } from 'react-router-dom';
      |                        ^^^^^^^^^^^^^^^^^^
   43 | (window as any).Vaadin ??= {};
   44 | (window as any).Vaadin.copilot ??= {};
   45 | (window as any).Vaadin.copilot._ref ??= {};
```

After this I ran `mvn vaadin:clean-frontend` and restarted the app. App runs without errors and I confirmed vaadin-router is being used.

### Expected behavior

After toggling `react.enable` to `false`, Flow should automatically figure out what to do to make the application run without frontend build failures.

### Minimal reproducible example

See description for instructions

### Versions

- Vaadin version: 24.4.11

**Comment by mshabarov** [comment]
We may try to extract a subset of config parameters that potentially affect frontend compilation (where the `react.enable` is among them) and for this group of parameters Flow always deletes `frontend/generated`. On the other hand, files in `jar-resources` are not necessarily need an update after configuration change, but rather after adding/removing JAR dependencies to the project.

**Comment by tltv** [comment]
Copilot adds that import to `vaadin.ts` when Flow runs `TaskGenerateBootstrap`. `TaskGenerateBootstrap` runs before `TaskRunNpmInstall`. Copilot checks if `react-router-dom` module exists in the `node_modules` and if it does, then it adds this import. It's not checking if react is enabled via `options.isReactEnabled()`. I think it should. @Artur- could you check if this can be fixed in `CopilotLoader`?

In more detail, once you build vaadin app with react enabled, it naturally installs `react-router-dom` module. If you run app again after disabling react, it's not installed anymore in the end as expected. Second run and following runs will work again because now Copilot is not adding imports automatically due to `react-router-dom` doesn't exist any more.

I can't think of any way to prevent this in Flow other than some very dirty hack that removes any attempt of adding common react imports via `TypeScriptBootstrapModifier` when react is disabled. But that would not prevent adding react imports anyway via non-common imports.

**Comment by tltv** [comment]
Fixed in Vaadin 24.5.0.beta1 release.
Waiting for release for Vaadin 24.4.13 (with Copilot 24.4.14).

**Comment by tltv** [comment]
Fix is released with Vaadin 24.4.13.
